modules/tensor_mechanics/test/tests/2D_slipweakening/tpv2052D.i

```diff
@@ -0,0 +1,257 @@
+# Verification of Benchmark Problem TPV205-2D from the SCEC Dynamic Rupture Validation exercises #
+# Reference: #
+# Harris, R. M.-P.-A. (2009). The SCEC/USGS Dynamic Earthquake Rupture Code Verification Exercise. Seismological Research Letters, vol. 80, no. 1, pages 119-126. #
+
+[Mesh]
+  [./msh]
+    type = GeneratedMeshGenerator
+    dim = 2
+    nx = 60
+    ny = 60
+    xmin = -3000
+    xmax = 3000
+    ymin = -3000
+    ymax = 3000
+  []
+  [./new_block]
+    type = ParsedSubdomainMeshGenerator
+    input = msh
+    combinatorial_geometry = 'y<0'
+    block_id = 1
+  []
+  [./split]
+    type = BreakMeshByBlockGenerator
+    input = new_block
+    split_interface = true
+  []
+[]
+
+[GlobalParams]
+  #primary variables
+  displacements = 'disp_x disp_y'
+  #damping ratio
+  q = 0.1
+  #characteristic length (m)
+  Dc = 0.4
+  #initial normal stress (Pa)
+  T2_o = 120e6
+  #dynamic friction coefficient
+  mu_d = 0.525
+  #element edge length (m)
+  len = 100
+[]
+
+[AuxVariables]
+  [./resid_x]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+  [./resid_y]
+      order = FIRST
+      family = LAGRANGE
+  []
+  [./resid_slipweakening_x]
+      order = FIRST
+      family = LAGRANGE
+  [../]
+  [./resid_slipweakening_y]
+      order = FIRST
+      family = LAGRANGE
+  [../]
+  [./disp_slipweakening_x]
+      order = FIRST
+      family = LAGRANGE
+  []
+  [./disp_slipweakening_y]
+      order = FIRST
+      family = LAGRANGE
+  []
+  [./mu_s]
+      order = CONSTANT
+      family = MONOMIAL
+  []
+  [./ini_shear_stress]
+      order = CONSTANT
+      family = MONOMIAL
+  []
+  [./tangent_jump_rate]
+      order = CONSTANT
+      family = MONOMIAL
+  []
+[]
+
+[Modules/TensorMechanics/CohesiveZoneMaster]
+  [./czm_ik]
+    boundary = 'Block0_Block1'
+    strain = SMALL
+    generate_output='traction_x traction_y jump_x jump_y normal_traction tangent_traction normal_jump tangent_jump'
+  [../]
+[]
+
+
+[Modules]
+  [./TensorMechanics]
+    [./Master]
+      [./all]
+        strain = SMALL
```

Comment:
Does this cohesive zone model only work for small strain? if it works for finite strain, it needs a test like the other cohesive zone models that show rotation. If it only works for small strain then the code needs to a warning or error if large strain is used.

Reply:
Yes, currently it only support small strain, I wonder how should I add this error (like moose error) to CZM, is there an example? @GiudGiud Please also help if you know this, thank you both

Reply:
check the strain model using whatever attribute stores it, then:
- use a paramError("strain" ...)  if in the TM action
- else use a MooseError with an informative error message

Reply:
Hi @GiudGiud, I would need further guidance on this.. I checked CohesiveZoneAction.C, it has "_strain" defined:

`_strain(getParam<MooseEnum>("strain").getEnum<Strain>()),`

Is there possible I could retrieve it in the SlipWeakening and use mooseError, I tried something but it didn't work, thanks!

Reply:
is SlipWeakening a derived class of CZM?

Reply:
it seems so.
in the constructor of the SlipWeaking you can look at the _strain attribute (unless it s private but doubt it)